### PR TITLE
Use k8s registry lib

### LINKED ISSUE #184: Replace etcd cesapp-lib regsistry with k8s-registry-lib to store configs in k8s natives

### PR DESCRIPTION
**Pull request by nfranzeck**
Resolves #184

### CODE REVIEW

File: controllers/resource/requirementsUpdater.go [hunk 1]
```diff
@@ -4,87 +4,67 @@ import (
...
-		select {
-		case <-ctx.Done():
-			return nil
-		case <-triggerChannel:
-			err := hlu.triggerSync(ctx)
-			if err != nil {
-				return err
-			}
+func (hlu *RequirementsUpdater) startWatch(ctx context.Context) error {
+	ctrl.LoggerFrom(ctx).Info(fmt.Sprintf("Start watching on global config for certificate key [%s]", triggerSyncKey))
```
[the remark this answers is not in the document]
Comment (nfranzeck) [reply controllers/resource/requirementsUpdater.go:47]: This is the adjusted log taken from current dev...but when i take a look at the implementation, this is just a trigger for sync, i will adjust this.

File: controllers/resource/volumeGenerator.go [hunk 2]
```diff
@@ -127,11 +122,44 @@ func createStaticVolumes(doguResource *k8sv1.Dogu) []corev1.Volume {
...
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: globalConfig},
+			},
+		},
+	}
+
+	doguConfigName := func(doguName string) string {
+		return fmt.Sprintf("%s-config", doguName)
+	}
```
[the remark this answers is not in the document]
Comment (nfranzeck) [reply controllers/resource/volumeGenerator.go:136]: the function is used twice for normalConfigVolume and sensitiveConfigVolume - so I created an internal function for it

[the remark this answers is not in the document]
Comment (kahoona77) [reply controllers/resource/volumeGenerator.go:136]: But it is used with the same doguName (`doguResource.Name`)

File: controllers/doguInstallManager.go [hunk 3]
```diff
@@ -28,48 +28,50 @@ const k8sDoguOperatorFieldManagerName = "k8s-dogu-operator"
...
+		execPodFactory:          exec.NewExecPodFactory(client, mgrSet.RestConfig, mgrSet.CommandExecutor),
+		doguConfigRepository:    configRepos.DoguConfigRepository,
+		sensitiveDoguRepository: configRepos.SensitiveDoguRepository,
 	}
 }
 
 // Install installs a given Dogu Resource. This includes fetching the dogu.json and the container image. With the
 // information Install creates a Deployment and a Service
-func (m *doguInstallManager) Install(ctx context.Context, doguResource *k8sv1.Dogu) error {
+func (m *doguInstallManager) Install(ctx context.Context, doguResource *k8sv1.Dogu) (err error) {
```
Comment (cesmarvin) [inline controllers/doguInstallManager.go:71]: 🚫 **[checkstyle]** <funlen> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function 'Install' has too many statements (59 > 40)

File: controllers/doguInstallManager.go [hunk 4]
```diff
@@ -209,6 +215,58 @@ func (m *doguInstallManager) createDoguResources(ctx context.Context, doguResour
 	return nil
 }
 
+func (m *doguInstallManager) createConfigs(ctx context.Context, doguName string, logger logr.Logger) (func(error), error) {
+	var doguCfgAlreadyExists, sensitiveCfgAlreadyExists bool
+
+	cleanUp := func(err error) {
```
Comment (cesmarvin) [inline controllers/doguInstallManager.go:221]: 🚫 **[checkstyle]** <contextcheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function `createConfigs$1` should pass the context parameter

File: controllers/dogu_manager.go [hunk 5]
```diff
@@ -45,12 +43,7 @@ type DoguManager struct {
 }
 
 // NewDoguManager creates a new instance of DoguManager
-func NewDoguManager(client client.Client, ecosystemClient ecoSystem.EcoSystemV1Alpha1Interface, operatorConfig *config.OperatorConfig, cesRegistry cesregistry.Registry, eventRecorder record.EventRecorder) (*DoguManager, error) {
-	err := validateKeyProvider(cesRegistry.GlobalConfig())
-	if err != nil {
-		return nil, fmt.Errorf("failed to validate key provider: %w", err)
-	}
-
+func NewDoguManager(client client.Client, ecosystemClient ecoSystem.EcoSystemV1Alpha1Interface, operatorConfig *config.OperatorConfig, eventRecorder record.EventRecorder) (*DoguManager, error) {
```
Comment (cesmarvin) [inline controllers/dogu_manager.go:46]: 🚫 **[checkstyle]** <funlen> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function 'NewDoguManager' is too long (74 > 60)

File: controllers/serviceaccount/serviceaccount_creator.go [hunk 6]
```diff
@@ -269,42 +183,44 @@ func (c *creator) executeCommand(ctx context.Context, consumerDogu *core.Dogu, s
...
 
-	return publicKey, nil
+	return nil
 }
 
-func (c *creator) writeServiceAccounts(doguConfig registry.ConfigurationContext, serviceAccount core.ServiceAccount, saCreds map[string]string, publicKey *keys.PublicKey) error {
-	for key, value := range saCreds {
-		path := "/sa-" + serviceAccount.Type + "/" + key
+func writeConfig(ctx context.Context, senDoguCfg *config.DoguConfig, cfgRepo SensitiveDoguConfigRepository) error {
+	update, err := cfgRepo.Update(ctx, *senDoguCfg)
```
Comment (cesmarvin) [inline controllers/serviceaccount/serviceaccount_creator.go:206]: 🚫 **[checkstyle]** <staticcheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>SA4006: this value of `update` is never used

File: controllers/serviceaccount/serviceaccount_creator.go [hunk 7]
```diff
@@ -269,42 +183,44 @@ func (c *creator) executeCommand(ctx context.Context, consumerDogu *core.Dogu, s
...
+	if err != nil {
+		if config.IsConflictError(err) {
+			mergedCfg, lErr := cfgRepo.SaveOrMerge(ctx, *senDoguCfg)
```
Comment (cesmarvin) [inline controllers/serviceaccount/serviceaccount_creator.go:209]: 🚫 **[checkstyle]** <staticcheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>SA4006: this value of `mergedCfg` is never used

File: controllers/resource/volumeGenerator.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (nfranzeck) [reply controllers/resource/volumeGenerator.go:136]: ahhh I see, my fault... change is on the way

File: controllers/dogu_manager.go (hunk 5 above)
Comment (nfranzeck) [reply controllers/dogu_manager.go:46]: Is see no option to fix this.

File: CHANGELOG.md [hunk 8]
```diff
@@ -5,6 +5,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+**Breaking Change ahead**
+### Removed
+- [#184] Remove support for internal ETCD
+
+### Changed
+- [#184] Add k8s-registry lib in version 0.2.2 to use config maps for configuration instead of the etcd.
+  - This change requires all other installed dogus to use doguctl >= v0.12.0
```
Comment (kahoona77) [inline CHANGELOG.md:14]:
```suggestion
  - This change requires all other installed dogus to use doguctl >= v0.12.1
```

File: Makefile [hunk 9]
```diff
@@ -1,9 +1,9 @@
 # Set these to the desired values
 ARTIFACT_ID=k8s-dogu-operator
-VERSION=1.2.0
+VERSION=1.3.0
```
Comment (kahoona77) [inline Makefile:3]:
```suggestion
VERSION=1.2.0
```

File: k8s/helm/values.yaml [hunk 10]
```diff
@@ -4,7 +4,7 @@ additionalImages:
 controllerManager:
   image:
     repository: cloudogu/k8s-dogu-operator
-    tag: 1.2.0
+    tag: 1.3.0
```
Comment (kahoona77) [inline k8s/helm/values.yaml:7]:
```suggestion
    tag: 1.2.0
```

**Review by kahoona77 — APPROVED** [review]
LGTM

**Pull request merged**